SimPEG/dask/electromagnetics/time_domain/simulation.py

```diff
@@ -155,265 +155,296 @@ def evaluate_receiver(source, receiver, mesh, time_mesh, fields):
 
-    # Check which time steps we need to compute
-    simulation_times = np.r_[0, np.cumsum(self.time_steps)] + self.t0
-    data_times = self.survey.source_list[0].receiver_list[0].times
-    n_times = len(data_times)
+                # Condition to start a new block
+                if (row_count + chunk_size) > (data_block_size * cpu_count()):
+                    row_count = 0
+                    block_count += 1
+                    blocks[block_count] = {}
 
-    for tInd, dt in tqdm(zip(reversed(range(self.nT)), reversed(self.time_steps))):
-        AdiagTinv = Ainv[dt]
-        Asubdiag = self.getAsubdiag(tInd)
-        row_count = 0
-        row_blocks = []
-        field_derivs = {}
-        source_blocks = []
-        d_count = 0
+                blocks[block_count][(s_id, r_id, ind)] = chunk, np.arange(
+                    row_index, row_index + chunk_size
+                ).astype(int)
+                row_index += chunk_size
+                row_count += chunk_size
 
-        data_bool = data_times > simulation_times[tInd]
+    return blocks
 
-        if data_bool.sum() == 0:
-            continue
 
-        # tc_loop = time()
-        # print(f"Loop sources for {tInd}")
-        for isrc, src in enumerate(self.survey.source_list):
-            column_inds = np.kron(
-                np.ones(int(src.vnD / n_times), dtype=bool), data_bool
-            )
-            # for block in range(len(self.field_derivs[tInd][isrc])):
-            if isrc not in field_derivs_t:
-                field_derivs[(isrc, src)] = self.field_derivs[tInd + 1][isrc].toarray()[
-                    :, column_inds
-                ]
-            else:
-                field_derivs[(isrc, src)] = field_derivs_t[isrc][:, column_inds]
-
-            n_data = self.field_derivs[tInd + 1][isrc].shape[1]
-            d_count += column_inds.sum()
-
-            if d_count > d_block_size:
-                source_blocks = block_append(
-                    self,
-                    f,
-                    AdiagTinv,
-                    field_derivs,
-                    m_size,
-                    row_count,
-                    tInd,
-                    solution_type,
-                    Jmatrix,
-                    Asubdiag,
-                    source_blocks,
-                    data_bool,
-                )
-                field_derivs = {}
-                row_count = d_count
-                d_count = 0
-
-        if field_derivs:
-            source_blocks = block_append(
-                self,
-                f,
-                AdiagTinv,
-                field_derivs,
-                m_size,
-                row_count,
-                tInd,
-                solution_type,
-                Jmatrix,
-                Asubdiag,
-                source_blocks,
-                data_bool,
-            )
+@delayed
+def deriv_block(
+    s_id, r_id, b_id, ATinv_df_duT_v, Asubdiag, local_ind, sub_ind, simulation, tInd
```

Comment:
..yeah ok, maybe leave the matrix names the way they are - looks like it's not possible to give everything a meaningful name.